cmd/jujud/agent/machine.go

```diff
@@ -985,6 +986,17 @@ func (a *MachineAgent) startStateWorkers(st *state.State) (worker.Worker, error)
 			a.startWorkerAfterUpgrade(runner, "mongoupgrade", func() (worker.Worker, error) {
 				return newUpgradeMongoWorker(st, a.machineId, a.maybeStopMongo)
 			})
+			a.startWorkerAfterUpgrade(runner, "statemetrics", func() (worker.Worker, error) {
+				return worker.NewSimpleWorker(func(stop <-chan struct{}) error {
```

Comment:
Would it not be neater if it was a method call? rather than defining new worker inline? 
It would be nice to test it in isolation as well.